src/main/java/wolox/training/controllers/UsersController.java

```diff
@@ -0,0 +1,85 @@
+package wolox.training.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNonOwnedException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.UserIdMismatchException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.Users;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UsersRepository;
+
+@RestController
+@RequestMapping("/api/users")
+public class UsersController {
+
+    @Autowired
+    private UsersRepository usersRepository;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @GetMapping("/username/{username}")
+    public Users findByUsername(@PathVariable String username) throws UserNotFoundException {
+        return usersRepository.findByUsername(username).orElseThrow(UserNotFoundException::new);
+    }
+
+    @GetMapping("{id}")
+    public Users get(@PathVariable Long id) throws UserNotFoundException {
+        return usersRepository.findById(id).orElseThrow(UserNotFoundException::new);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Users create(@RequestBody Users users) {
+        return usersRepository.save(users);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) throws UserNotFoundException {
+        usersRepository.findById(id)
+            .orElseThrow(UserNotFoundException::new);
+        usersRepository.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public Users update(@RequestBody Users user, @PathVariable Long id)
+        throws UserIdMismatchException, UserNotFoundException {
+        if (user.getId() != id)
```

Comment:
You can use `.equals` instead of `!=` since you're working with classes (Long) rather than primitive types.